tcms/xmlrpc/api/build.py

```diff
@@ -25,7 +25,10 @@ def filter(query={}):
         :return: List of serialized :class:`tcms.management.models.Build` objects
         :rtype: list(dict)
     """
-    return Build.to_xmlrpc(query)
+
+    if query:
+        return Build.to_xmlrpc(query)
```

Comment:
it will be clearer if you write it like this:
```
if query is None:
        query = {} 
return Build.to_xmlrpc(query)
```

This way you have a single point of exit, and also it is more clear what you are doing.

Reply:
+1